test/images/netproxy/netproxy.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	httpPort  = 8080
+	udpPort   = 8081
+	shellPath = "/bin/sh"
+)
+
+type output struct {
+	responses []string
+	errors    []string
+}
+
+func init() {
+	flag.IntVar(&httpPort, "http-port", 8080, "HTTP Listen Port")
+	flag.IntVar(&udpPort, "udp-port", 8081, "UDP Listen Port")
+}
+
+func main() {
+	flag.Parse()
+	go startUDPServer(udpPort)
+	startHTTPServer(httpPort)
+}
+
+func startHTTPServer(httpPort int) {
+	http.HandleFunc("/shutdown", shutdownHandler)
+	http.HandleFunc("/hostName", hostNameHandler)
+	http.HandleFunc("/shell", shellHandler)
+	http.HandleFunc("/dial", dialHandler)
+	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", httpPort), nil))
+}
+
+func shutdownHandler(w http.ResponseWriter, r *http.Request) {
+	os.Exit(0)
+}
+
+func dialHandler(w http.ResponseWriter, r *http.Request) {
+	values, err := url.Parse(r.URL.RequestURI())
+	if err != nil {
+		http.Error(w, fmt.Sprintf("%v", err), http.StatusBadRequest)
+		return
+	}
+
+	host := values.Query().Get("host")
+	port := values.Query().Get("port")
+	request := values.Query().Get("request") // hostName
+	protocol := values.Query().Get("protocol")
```

Comment:
is this "tcp/udp" or "http/https"?  If the latter, maybe "scheme" is a better name